src/components/FlowList.vue

```diff
@@ -0,0 +1,158 @@
+            <p-select v-model="filter.sort" :options="flowSortOptions" />
+            <p-tags-input v-model="filter.flowRuns.tags.name" :placeholder="localization.info.addTagPlaceholder" class="flow-list__flow-run-tags">
+              <template #empty-message>
+                <span class="flow-list__flow-run-tags--empty">{{ localization.info.filterByFlowRunTags }}</span>
+              </template>
+            </p-tags-input>
+          </div>
+        </slot>
+      </template>
+
+      <p-virtual-scroller
+        :items="flows"
+        :chunk-size="20"
+        :item-estimate-height="140"
+        item-key="id"
+      >
+        <template #default="{ item }">
+          <slot v-bind="{ item, selected, disabled, update: handleUpdate, delete: handleDelete }">
+            <FlowListItem
+              v-model:selected="selected"
+              :value="item.id"
+              :flow="item"
+              :filter="props.filter"
+              :disabled="disabled"
+              class="flow-list__flow"
+              @update="handleUpdate"
+              @delete="handleDelete"
+            />
+          </slot>
+        </template>
+      </p-virtual-scroller>
+
+      <template #footer-end>
+        <slot name="footer-end" v-bind="{ page, pages }">
+          <p-pager v-if="pages > 1" v-model:page="page" :pages="pages" />
+        </slot>
+      </template>
+    </p-layout-table>
+  </div>
+</template>
+
+<script lang="ts" setup>
+  import { useDebouncedRef, useSubscription } from '@prefecthq/vue-compositions'
+  import { computed, ref } from 'vue'
+  import { FlowListItem, FlowsDeleteButton, ResultsCount, SearchInput, SelectedCount } from '@/components'
+  import { useFlowsFilterFromRoute, useWorkspaceApi } from '@/compositions'
+  import { localization } from '@/localization'
+  import { FlowsFilter } from '@/models/Filters'
+  import { flowSortOptions } from '@/types/SortOptionTypes'
+
+  const props = defineProps<{
+    filter?: FlowsFilter,
+    disabled?: boolean,
+  }>()
+
+  const emit = defineEmits<{
+    (event: 'delete' | 'update', value?: string): void,
+  }>()
+
+  const DEFAULT_LIMIT = 40
+
+  const api = useWorkspaceApi()
+  const search = ref<string>('')
+  const searchDebounced = useDebouncedRef(search, 800)
+
+  const page = ref(1)
+  const offset = computed({
+    get: () => (page.value - 1) * DEFAULT_LIMIT,
+    set: (value: number) => {
+      page.value = Math.ceil(value / DEFAULT_LIMIT) + 1
+    },
+  })
+  const pages = computed(() => Math.ceil((flowsCount.value ?? DEFAULT_LIMIT) / DEFAULT_LIMIT))
+
+  const { filter } = useFlowsFilterFromRoute({
+    ...props.filter,
+    flows: {
+      ...props.filter?.flows,
+      nameLike: searchDebounced,
+    },
+    offset,
+    limit: DEFAULT_LIMIT,
+  })
+
+  const selected = ref<string[]>([])
+
+  const flowsSubscription = useSubscription(api.flows.getFlows, [filter])
+  const flows = computed(() => flowsSubscription.response ?? [])
```

Comment:
There's a `useFlows` composition for this. It currently doesn't do the `?? []` piece but I think it probably should.